tests/mocks/MockHttpRequester.hh

```diff
@@ -0,0 +1,39 @@
+#ifndef RA_SERVICES_MOCK_HTTPREQUESTER_HH
+#define RA_SERVICES_MOCK_HTTPREQUESTER_HH
+#pragma once
+
+#include "services\IHttpRequester.hh"
+#include "services\ServiceLocator.hh"
+
+namespace ra {
+namespace services {
+namespace mocks {
+
+class MockHttpRequester : public IHttpRequester
+{
+public:
+    explicit MockHttpRequester(std::function<Http::Response(const Http::Request&)>&& fHandler) noexcept
```

Comment:
I don't think this can be `noexcept`.

Reply:
It's not that the compiler wouldn't complain, but std::function's copy constructor might throw an exception. It's not a big deal.

Reply:
The purpose is to identify where exception handling will need to be targeted.

One of the rulesets; `Function Rules`, will warn on this but hasn't been applied yet (done but holding off on doing a PR for it).
Example:
Also: `fHandler `was never moved but it's move constructor might throw.
`explicit MockHttpRequester(std::function<Http::Response(const Http::Request&)>&& fHandler) noexcept(std::is_nothrow_move_constructible_v<std::function<Http::Response(const Http::Request&)>>);`

I guarantee that it will not be noexcept (hover over the constructor name and the `noexcept `will not be there) if you do that since it's not. Don't worry about it now, we'll handle it when the time comes.